solvation_analysis/analysis.py

```diff
@@ -0,0 +1,144 @@
+from MDAnalysis.lib.distances import capped_distance
+import numpy as np
+from solvation_analysis.rdf_parser import identify_solvation_cutoff
+from solvation_analysis.analysis_library import (
+    _CoordinationNumber,
+    _Pairing,
+    _IonSpeciation,
+    _SolvationData
+)
+from solvation_analysis.solvation import get_radial_shell, get_closest_n_mol
+
+
+class Solution(AnalysisBase):
+    """
+    The core class of the solvation module.
+    """
+
+    def __init__(
+        self,
+        solute,
+        solvents,
+        radii=None,
+        rdf_kernel=None,
+        kernel_kwargs=None,
+        rdf_kwargs=None,
+        **kwargs,
+    ):
+        """
+        Parameters
+        ----------
+        solute (AtomGroup): an atom group  # TODO: force to atom group
+        solvents (dict): a dictionary of names and atom groups.
+            e.g. {"name_1": solvent_group_1, "name_2": solvent_group_2, ...}
+        radii (dict): an optional dictionary of solvation radii, any radii not
+            given will be calculated. e.g. {"name_2": radius_2, "name_5": radius_5}
+        rdf_kernel (func): this function must take rdf bins and data as input and return
+            a solvation radius as output. e.g. rdf_kernel(bins, data) -> 3.2. By default,
+            the rdf_kernel is solvation_analysis.rdf_parser.identify_solvation_cutoff.
+        kernel_kwargs: kwargs passed to rdf_kernel
+        kernel_kwargs: kwargs passed to inner rdf
+        kwargs: kwargs passed to AnalysisBase
+        """
+        super(Solution, self).__init__(solute.universe.trajectory, **kwargs)
+        self.radii = {} if radii is None else radii
+        self.kernel = identify_solvation_cutoff if rdf_kernel is None else rdf_kernel
+        # if not kernel:
+        #     self.kernel = identify_solvation_cutoff
+        self.kernel_kwargs = {} if kernel_kwargs is None else kernel_kwargs
+        self.rdf_kwargs = {"range": (0, 8.0)} if rdf_kwargs is None else rdf_kwargs
+        # if not kernel_kwargs:
+        #     self.kernel_kwargs = {}
+
+        self.solute = solute
+        self.solvents = solvents
+        self.u = self.solute.universe
+        self.rdf_plots = {}
+        self.rdf_data = {}
+        self.ion_speciation = None
+        self.ion_pairing = None
+        self.coordination_numbers = None
+        self.solvation_frames = []
+
+    @classmethod
+    def _plot_solvation_radius(cls, bins, data, radius):
+        fig, ax = plt.subplots()
+        ax.plot(bins, data, "b-", label="rdf")
+        ax.axvline(radius, color="r", label="solvation radius")
+        ax.set_xlabel("Radial Distance (A)")
+        ax.set_ylabel("Probability Density")
+        ax.legend()
+        return fig, ax
+
+    def _prepare(self):
+        for name, solvent in self.solvents.items():
+            # generate and save RDFs
+            rdf = InterRDF(self.solute, solvent, **self.rdf_kwargs)
+            # TODO: specify start and stop args
+            rdf.run()
```

Comment:
Should rdf_kwargs here be fed to `InterRDF` and `run`? Looking at the signatures I am guessing I should pass only to `run`, the opposite of what I am doing here?

Maybe I'm wrong, but I don't think its possible to have kwargs for both functions in the same dict?

Reply:
I could include two sets of rdf_kwargs but that may be confusing. I could also filter the dict to select for specific keywords relevant to one or the other function.

Reply:
The cleaner approach is to have init_kwargs and run_kwargs. Are there any kwargs with the same name that can be used for either? In any case, given that your code orchestrates the workflow, you should sort out in advance what goes where. I'd be careful to just pass everything from the user side through unless you're sure that it will not interfere with how your code wants to run RDF.

Reply:
There is no overlap in the kwarg names. I am going with @orbeckst's approach as it seems cleanest. I only included the init_kwargs in the name of flexibility, I doubt anyone would have much use for them. Should I just leave them out in that case?

Reply:
Specifically, init_kwargs would be the range and n_bins for the rdf. I am comfortable with the defaults I have chosen. Think they would be worth including?

Reply:
@orbeckst's solution is a good one. Good also to have flexibility if it doesn't add too much complexity. As it is impossible to imagine every user use case, i feel having the flexibility to play with your `RDF` kwargs if you need to is a good thing.

Reply:
What I am trying to say with that confusing sentence is that i think you should also have `init_kwargs`

Reply:
Fixed this as suggested. I'm going to resolve.